engine/src/main/java/org/hibernate/search/engine/environment/bean/BeanResolver.java

```diff
@@ -11,74 +11,77 @@
 import java.util.List;
 import java.util.function.Function;
 
-import org.hibernate.search.engine.environment.bean.spi.BeanResolver;
+import org.hibernate.search.engine.environment.bean.spi.BeanProvider;
 import org.hibernate.search.util.common.SearchException;
 import org.hibernate.search.util.common.impl.Contracts;
 import org.hibernate.search.util.common.impl.SuppressingCloser;
 
 /**
- * The main entry point for components looking to retrieve user-provided beans.
+ * The main entry point for components looking to resolve a bean reference into a (usually user-provided) bean.
  * <p>
- * Depending on the integration, beans may be resolved using reflection (expecting a no-argument constructor),
- * or using a more advanced dependency injection context (CDI, Spring DI).
+ * Depending on the integration, beans may be instantiated using reflection (expecting a no-argument constructor),
+ * or provided by a more advanced dependency injection context (CDI, Spring DI).
  * <p>
- * Regardless of the implementations, this interface is used to retrieve the beans,
- * referenced either by their name, by their type, or both.
+ * Regardless of the underlying implementation, this interface is used to resolve beans,
+ * referenced either
+ * {@link #resolve(Class) by their type},
+ * or {@link #resolve(Class, String) by their type and name},
+ * or {@link #resolveRole(Class) by their role}.
  * <p>
- * This interface may be used by any Hibernate Search module,
- * but should only be implemented by the Hibernate Search engine itself;
- * if you are looking for implementing your own bean resolver,
- * you should implement {@link BeanResolver} instead.
+ * This interface is API,
+ * but should only be implemented by Hibernate Search itself;
```

Comment:
Then maybe we should move it from a spi to an impl package.

Reply:
I was trying to say that even though users can *call* the methods of this interface, nobody should not try to *implement* this interface. If people want to customize the bean retrieval behavior, they should implement `BeanProvider` instead (as explained below).